src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -0,0 +1,624 @@
+				// we'll multiple saturation and value by 100 to put them in the range of 0-100 instead of 0-1.
+				newHsv.S *= 100;
+				newHsv.V *= 100;
+
+				// Uno Doc: *valueToIncrement replaced with ref local variable for C#, must be initialized
+				ref double valueToIncrement = ref newHsv.H;
+				double incrementAmount = 0.0;
+
+				// If we're adding a small increment, then we'll just add or subtract 1.
+				// If we're adding a large increment, then we want to snap to the next
+				// or previous major value - for hue, this is every increment of 30;
+				// for saturation and value, this is every increment of 10.
+				switch (channel)
+				{
+					case ColorPickerHsvChannel.Hue:
+						valueToIncrement = ref newHsv.H;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 30;
+						break;
+
+					case ColorPickerHsvChannel.Saturation:
+						valueToIncrement = ref newHsv.S;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					case ColorPickerHsvChannel.Value:
+						valueToIncrement = ref newHsv.V;
+						incrementAmount = amount == IncrementAmount.Small ? 1 : 10;
+						break;
+
+					default:
+						throw new InvalidOperationException("Invalid ColorPickerHsvChannel."); // Uno Doc: 'winrt::hresult_error(E_FAIL);'
+				}
+
+				double previousValue = valueToIncrement;
+
+				valueToIncrement += (direction == IncrementDirection.Lower ? -incrementAmount : incrementAmount);
+
+				// If the value has reached outside the bounds, we were previous at the boundary, and we should wrap,
+				// then we'll place the selection on the other side of the spectrum.
+				// Otherwise, we'll place it on the boundary that was exceeded.
+				if (valueToIncrement < minBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == minBound) ? maxBound : minBound;
+				}
+
+				if (valueToIncrement > maxBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == maxBound) ? minBound : maxBound;
+				}
+
+				// We multiplied saturation and value by 100 previously, so now we want to put them back in the 0-1 range.
+				newHsv.S /= 100;
+				newHsv.V /= 100;
+			}
+			else
+			{
+				// While working with named colors, we're going to need to be working in actual HSV units,
+				// so we'll divide the min bound and max bound by 100 in the case of saturation or value,
+				// since we'll have received units between 0-100 and we need them within 0-1.
+				if (channel == ColorPickerHsvChannel.Saturation ||
+					channel == ColorPickerHsvChannel.Value)
+				{
+					minBound /= 100;
+					maxBound /= 100;
+				}
+
+				newHsv = FindNextNamedColor(originalHsv, channel, direction, shouldWrap, minBound, maxBound);
+			}
+
+			return newHsv;
+		}
+
+		// Uno Doc: This function is removed and instead Math.Sign() is used.
+		// Implementation of the signum function, which returns the sign of a number while discarding its value.
+		/*template <typename T>
+		int sgn(T val)
+		{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)